packages/components/image.tsx

```diff
@@ -1,176 +1,69 @@
-import { Connect, Package } from "frontity/types";
-import useInView from "@frontity/hooks/use-in-view";
-
-// Hides any image rendered by this component that is not
-// inside a <noscript> when JS is disabled.
-const noJsStyles = `
-  :not(noscript) > .frontity-lazy-image {
-    display: none;
-  }
-`;
-
-// Finds all the images rendered by this component and maps
-// their `data-src` and `data-srcset` attributes to `src` and `srcset`
-// when the browser doesn't support Proxy or IntersectionObserver.
-const noProxyScript = `
-  if (typeof window !== "undefined" && (!("Proxy" in window) || !("IntersectionObserver" in window))) {
-    document.addEventListener("DOMContentLoaded", function() {
-      var images = document.querySelectorAll("img.frontity-lazy-image");
-      for (i = 0; i < images.length; ++i) {
-        var image = images[i];
-        image.setAttribute("src", image.getAttribute("data-src"));
-        image.setAttribute("srcset", image.getAttribute("data-srcset"));
-        image.removeAttribute("data-src");
-        image.removeAttribute("data-srcset");
-        image.removeAttribute("style");
-      }
-    });
-  }
-`;
-
-export interface Props {
-  src?: string;
-  srcSet?: string;
-  sizes?: string;
-  alt?: string;
+import { connect, useConnect } from "frontity";
+import { Frontity, Connect } from "frontity/types";
+
+/**
+ * Props for the {@link Image} component.
+ */
+export interface ImageProps extends React.ImgHTMLAttributes<HTMLImageElement> {
+  /**
+   * CSS class(es).
+   */
   className?: string;
-  rootMargin?: string;
-  loading?: "lazy" | "eager";
-  height?: number;
-}
-
-type Image = React.FC<Connect<Package, Props>>;
 
-interface Attributes extends Props {
-  "data-src"?: string;
-  "data-srcset"?: string;
-  style?: { visibility: "hidden" };
+  /**
+   * The `rootMargin` of useInView.
+   *
+   * @deprecated This property was used in the previous version of the Image
+   * component and has no effect anymore as the Image component relies only on
+   * native lazy-loading.
+   */
+  rootMargin?: number;
 }
 
-type NoScriptImage = React.FC<Attributes>;
-
-interface ChangeAttributes {
-  (attrs: Attributes): Attributes;
-}
-
-const changeAttributes: ChangeAttributes = (attrs) => {
-  const attributes = { ...attrs };
-
-  attributes.src = attributes["data-src"];
-  attributes.srcSet = attributes["data-srcset"];
-  delete attributes["data-src"];
-  delete attributes["data-srcset"];
-  delete attributes["style"];
-
-  return attributes;
-};
-
-const NoScriptImage: NoScriptImage = (props) => {
-  const attributes = { ...props };
-
-  return (
-    <noscript>
-      <img {...attributes} />
-    </noscript>
-  );
-};
-
-const Image: Image = ({
-  state,
-  alt,
-  src,
-  srcSet,
-  sizes,
-  className,
+/**
+ * The Image component.
+ *
+ * @param props - Defined in {@link ImageProps}.
+ * @returns A React component.
+ */
+const Image: React.FC<Connect<Frontity, ImageProps>> = ({
```

Comment:
When using `connect` with `injectProps: false` you don't need to use `Connect` to type the component, because it doesn't receive the store 🙂

```suggestion
const Image: React.FC<ImageProps> = ({ 
```